icechunk/src/session.rs

```diff
@@ -199,6 +248,8 @@ pub struct Session {
     snapshot_id: SnapshotId,
     change_set: ChangeSet,
     default_commit_metadata: SnapshotProperties,
+    // This is an optimization so that we needn't figure out the split sizes on every set.
+    splits: HashMap<NodeId, ManifestSplits>,
```

Comment:
Stored on Session since we'll need to grab splits for arrays created in a previous session, but being modified in this one & the change_set doesn't have enough info (it receives NodeId, not NodeSnapshot IIRC)